pkg/api/errors/errors.go

```diff
@@ -297,6 +297,17 @@ func NewGenericServerResponse(code int, verb, kind, name, serverMessage string,
 	case len(kind) > 0:
 		message = fmt.Sprintf("%s (%s %s)", message, strings.ToLower(verb), kind)
 	}
+	var causes []api.StatusCause
+	if isUnexpectedResponse {
+		causes = []api.StatusCause{
+			{
+				Type:    api.CauseTypeUnexpectedServerResponse,
+				Message: serverMessage,
+			},
+		}
+	} else {
+		causes = nil
```

Comment:
Yes NewNotFound is different: https://github.com/GoogleCloudPlatform/kubernetes/blob/3565a0861c62a827fc6112968262c75c8c59d5d1/pkg/api/errors/errors.go#L91
It has a bit more details.
Example output from NewNotFound:
```
$curl -X GET -i http://localhost:8082/api/v1beta3/namespaces/default/services/kube
HTTP/1.1 404 Not Found
Content-Type: application/json
Date: Thu, 16 Apr 2015 22:06:17 GMT
Content-Length: 232

{
  "kind": "Status",
  "apiVersion": "v1beta3",
  "metadata": {},
  "status": "Failure",
  "message": "service \"kube\" not found",
  "reason": "NotFound",
  "details": {
    "id": "kube",
    "kind": "service"
  },
  "code": 404
}
```

Example output from go-restful 404:
```
curl -X GET -i http://localhost:8082/api/v1beta3/redirect/namespaces/default/servfds/fdsfsd
HTTP/1.1 404 Not Found
Content-Type: application/json
Date: Thu, 16 Apr 2015 22:07:41 GMT
Content-Length: 211

{
  "kind": "Status",
  "apiVersion": "v1beta3",
  "metadata": {},
  "status": "Failure",
  "message": "the server could not find the requested resource",
  "reason": "NotFound",
  "details": {},
  "code": 404
}
```